File: lab/python/POO/sistema_notas.py
class Aluno:
    def __init__(self, nome):
        self.nome = nome
        self.notas = []

    def adicionar_nota(self, nota):
        self.notas.append(nota)

    def calcular_media(self):
        if not self.notas:
            return 0
        return sum(self.notas) / len(self.notas)

    def aprovado(self):
        return self.calcular_media() >= 6

    def __repr__(self):
        return (
            f"Aluno: {self.nome}, Notas: {self.notas}, "
            f"Média: {self.calcular_media():.2f}"
        )


if __name__ == "__main__":
    aluno1 = Aluno("Maria")
    aluno1.adicionar_nota(7)
    aluno1.adicionar_nota(8)
    aluno1.adicionar_nota(6)
    aluno1.calcular_media()
    print(aluno1)
    print(aluno1.aprovado())

    print()

    aluno2 = Aluno("João")
    aluno2.adicionar_nota(5)
    aluno2.adicionar_nota(4)
    aluno2.adicionar_nota(6)
    aluno2.calcular_media()
    print(aluno2)
    print(aluno2.aprovado())

File: lab/javascript/POO/sistema_notas.js
class Aluno {
    constructor(nome) {
        this.nome = nome;
        this.notas = [];
    }

    adicionarNota(nota) {
        this.notas.push(nota);
    }

    calcularMedia() {
        if (this.notas.length === 0) return 0;
        let soma = this.notas.reduce((acc, n) => acc + n, 0);
        return soma / this.notas.length;
    }

    aprovado() {
        return this.calcularMedia() >= 6;
    }

    toString() {
        return `Aluno: ${this.nome}, Notas: [${this.notas.join(", ")}], Média: ${this.calcularMedia().toFixed(2)}`;
    }
}

// Testes
let aluno1 = new Aluno("Maria");
aluno1.adicionarNota(7);
aluno1.adicionarNota(8);
aluno1.adicionarNota(6);
console.log(aluno1.toString());
console.log(aluno1.aprovado());

console.log();

let aluno2 = new Aluno("João");
aluno2.adicionarNota(5);
aluno2.adicionarNota(4);
aluno2.adicionarNota(6);
console.log(aluno2.toString());
console.log(aluno2.aprovado());
